libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/MetadataSyncManager.java

```diff
@@ -0,0 +1,266 @@
+ * list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.smartsync.manager;
+
+import com.salesforce.androidsdk.accounts.UserAccount;
+import com.salesforce.androidsdk.app.SalesforceSDKManager;
+import com.salesforce.androidsdk.smartstore.store.IndexSpec;
+import com.salesforce.androidsdk.smartstore.store.QuerySpec;
+import com.salesforce.androidsdk.smartstore.store.SmartStore;
+import com.salesforce.androidsdk.smartsync.app.SmartSyncSDKManager;
+import com.salesforce.androidsdk.smartsync.model.Metadata;
+import com.salesforce.androidsdk.smartsync.target.MetadataSyncDownTarget;
+import com.salesforce.androidsdk.smartsync.target.SyncDownTarget;
+import com.salesforce.androidsdk.smartsync.util.Constants;
+import com.salesforce.androidsdk.smartsync.util.SmartSyncLogger;
+import com.salesforce.androidsdk.smartsync.util.SyncOptions;
+import com.salesforce.androidsdk.smartsync.util.SyncState;
+
+import org.json.JSONArray;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Provides an easy way to fetch metadata using {@link com.salesforce.androidsdk.smartsync.target.MetadataSyncDownTarget}.
+ * This class handles creating a soup, storing synched data and reading it into
+ * a meaningful data structure, i.e. {@link com.salesforce.androidsdk.smartsync.model.Metadata}.
+ *
+ * @author bhariharan
+ */
+public class MetadataSyncManager {
+
+    private static final String FEATURE_METADATA_SYNC = "MD";
+    static final String SOUP_NAME = "sfdcMetadata";
+    static final String QUERY = "SELECT {" + SOUP_NAME + ":_soup} FROM {" + SOUP_NAME +
+            "} WHERE {" + SOUP_NAME + ":" + Constants.ID + "} = '%s'";
```

Comment:
I see that you store the sobjectType in the Id field [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/MetadataSyncDownTarget.java#L92).
Should we just call that field sobjectType and specify "sobjectType" as the idFieldName in the sync down target?

Reply:
We did the same for the layout target (used `Id` field), so I kept it consistent with that since we're inserting a new field in both cases.

Reply:
Ok